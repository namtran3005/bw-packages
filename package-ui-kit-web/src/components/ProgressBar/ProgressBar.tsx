import * as React from 'react';
import styled from 'styled-components';
import { LinearProgress as MuiLinearProgress } from '@material-ui/core';
import { LinearProgressProps as MuiLinearProgressProps } from '@material-ui/core/LinearProgress';
import { ColorsType, Colors, Spacing } from '@bitwala-cryptobank-squad/package-theme';

interface RootProps extends MuiLinearProgressProps {
  foreground?: ColorsType;
  weight?: 'thin' | 'normal';
  innerRef?: any; // eslint-disable-line @typescript-eslint/no-explicit-any
}

const RootComponent: React.FunctionComponent<RootProps> = (props) => (
  <MuiLinearProgress
    classes={{
      root: 'root',
      colorPrimary: 'colorPrimary',
      colorSecondary: 'colorSecondary',
      buffer: 'buffer',
      query: 'query',
      dashed: 'dashed',
      dashedColorPrimary: 'dashedColorPrimary',
      dashedColorSecondary: 'dashedColorSecondary',
      bar: 'bar',
      barColorPrimary: 'barColorPrimary',
      barColorSecondary: 'barColorSecondary',
      bar1Indeterminate: 'bar1Indeterminate',
      bar2Indeterminate: 'bar2Indeterminate',
      bar1Determinate: 'bar1Determinate',
      bar1Buffer: 'bar1Buffer',
      bar2Buffer: 'bar2Buffer',
    }}
    {...props}
  />
);

const Root = styled(RootComponent)`
  &&& {
    border: none;
    height: ${Spacing.unit / 2}px;
    background-color: ${(props) =>
      props.foreground !== undefined
        ? Colors.getWithAlpha(props.foreground, 0.2)
        : Colors.getWithAlpha('primaryGreen', 0.2)};

    .bar {
      background-color: ${(props) =>
        Colors.get(
          props.foreground !== undefined ? props.foreground : 'primaryGreen'
        )};
    }
  }
`;

Root.displayName = 'LinearProgress';
RootComponent.displayName = 'LinearProgress';

export const ProgressBar: React.FC<RootProps> = (props) => <Root {...props} />;

export type ProgressBarProps = RootProps;
