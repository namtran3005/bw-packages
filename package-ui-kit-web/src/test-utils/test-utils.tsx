import {
  render as rtlRender,
  waitFor,
  RenderOptions as RTLRenderOptions,
} from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import userEvent from '@testing-library/user-event';
import { WebAppThemeProvider } from '../theme';
import '@testing-library/jest-dom';
import '@testing-library/jest-dom/extend-expect';

type RenderOptions = RTLRenderOptions;

function renderWithTestHarness(
  ui: React.ReactElement,
  renderOptions: RenderOptions = {}
) {
  function Wrapper({ children }: any) {
    return (
      <MemoryRouter>
        <WebAppThemeProvider>{children}</WebAppThemeProvider>
      </MemoryRouter>
    );
  }

  return rtlRender(ui, {
    wrapper: Wrapper,
    ...renderOptions,
  });
}

// Use if you're in a hurry (better than nothing)
// be sure to verify no memory leak from implicit
// waitFor by adding these flags to jest: --runInBand --detectOpenHandles
async function renderAct(ui: React.ReactElement, options: RenderOptions = {}) {
  const view = renderWithTestHarness(ui, options);
  return await waitFor(() => view, {
    timeout: 5000,
    onTimeout: (e) => {
      // eslint-disable-next-line no-console
      console.error(e);
      return e;
    },
  });
}

export * from '@testing-library/react';

export { renderWithTestHarness, renderAct, userEvent };
