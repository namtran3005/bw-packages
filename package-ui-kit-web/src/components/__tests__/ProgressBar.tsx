import * as React from 'react';

import { renderWithTestHarness } from '../../test-utils/test-utils';
import { ProgressBar } from '../ProgressBar';

const defaultProps = {};

describe('ProgressBar', () => {
  it('should render progress bar', () => {
    const { baseElement } = renderWithTestHarness(
      <ProgressBar {...defaultProps} />
    );
    expect(baseElement).toMatchSnapshot();
  });
  it('should render determinate progress bar', () => {
    const { baseElement } = renderWithTestHarness(
      <ProgressBar {...defaultProps} />
    );
    expect(baseElement).toMatchSnapshot();
  });
  it('should render indeterminate progress bar', () => {
    const { baseElement } = renderWithTestHarness(
      <ProgressBar {...defaultProps} />
    );
    expect(baseElement).toMatchSnapshot();
  });
  it('should render progress bar custom foreground', () => {
    const { baseElement } = renderWithTestHarness(
      <ProgressBar
        variant="determinate"
        foreground="success"
        {...defaultProps}
      />
    );
    expect(baseElement).toMatchSnapshot();
  });
});
