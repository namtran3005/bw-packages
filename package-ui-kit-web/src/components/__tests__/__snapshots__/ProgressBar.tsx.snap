// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProgressBar should render determinate progress bar 1`] = `
<body>
  .c0.c0.c0 {
  border: none;
  height: 4px;
  background-color: rgba(110,158,150,0.2);
}

.c0.c0.c0 .bar {
  background-color: #6E9E96;
}

<div>
    <div
      class="MuiLinearProgress-root root MuiLinearProgress-colorPrimary colorPrimary c0 MuiLinearProgress-indeterminate"
      role="progressbar"
    >
      <div
        class="MuiLinearProgress-bar bar MuiLinearProgress-barColorPrimary barColorPrimary MuiLinearProgress-bar1Indeterminate bar1Indeterminate"
      />
      <div
        class="MuiLinearProgress-bar bar MuiLinearProgress-bar2Indeterminate bar2Indeterminate MuiLinearProgress-barColorPrimary barColorPrimary"
      />
    </div>
  </div>
</body>
`;

exports[`ProgressBar should render indeterminate progress bar 1`] = `
<body>
  .c0.c0.c0 {
  border: none;
  height: 4px;
  background-color: rgba(110,158,150,0.2);
}

.c0.c0.c0 .bar {
  background-color: #6E9E96;
}

<div>
    <div
      class="MuiLinearProgress-root root MuiLinearProgress-colorPrimary colorPrimary c0 MuiLinearProgress-indeterminate"
      role="progressbar"
    >
      <div
        class="MuiLinearProgress-bar bar MuiLinearProgress-barColorPrimary barColorPrimary MuiLinearProgress-bar1Indeterminate bar1Indeterminate"
      />
      <div
        class="MuiLinearProgress-bar bar MuiLinearProgress-bar2Indeterminate bar2Indeterminate MuiLinearProgress-barColorPrimary barColorPrimary"
      />
    </div>
  </div>
</body>
`;

exports[`ProgressBar should render progress bar 1`] = `
.c0.c0.c0 {
  border: none;
  height: 4px;
  background-color: rgba(110,158,150,0.2);
}

.c0.c0.c0 .bar {
  background-color: #6E9E96;
}

<body>
  <div>
    <div
      class="MuiLinearProgress-root root MuiLinearProgress-colorPrimary colorPrimary c0 MuiLinearProgress-indeterminate"
      role="progressbar"
    >
      <div
        class="MuiLinearProgress-bar bar MuiLinearProgress-barColorPrimary barColorPrimary MuiLinearProgress-bar1Indeterminate bar1Indeterminate"
      />
      <div
        class="MuiLinearProgress-bar bar MuiLinearProgress-bar2Indeterminate bar2Indeterminate MuiLinearProgress-barColorPrimary barColorPrimary"
      />
    </div>
  </div>
</body>
`;

exports[`ProgressBar should render progress bar custom foreground 1`] = `
<body>
  .c0.c0.c0 {
  border: none;
  height: 4px;
  background-color: rgba(28,193,140,0.2);
}

.c0.c0.c0 .bar {
  background-color: #1CC18C;
}

<div>
    <div
      class="MuiLinearProgress-root root MuiLinearProgress-colorPrimary colorPrimary c0 MuiLinearProgress-determinate"
      foreground="success"
      role="progressbar"
    >
      <div
        class="MuiLinearProgress-bar bar MuiLinearProgress-barColorPrimary barColorPrimary MuiLinearProgress-bar1Determinate bar1Determinate"
      />
    </div>
  </div>
</body>
`;
